CHANGELOG.md

```diff
@@ -28,6 +28,8 @@ We use *breaking :warning:* to mark changes that are not backward compatible (re
 
 ### Removed
 
+- [#5426](https://github.com/thanos-io/thanos/pull/5426) Compactor: Remove flag `--block-sync-concurrency`.
```

Comment:
Maybe let's specify this was not being used, so it does not give impression this is a breaking change?

Reply:
Updated.